### PR TITLE
 Add moment map generation information to FITS header

### LINKED ISSUE #1024: Include moment map generation information
Raised in CARTA tutorial by Gary Fuller: 

We need to add some information to the FITS header history about moment map generation. For example, the velocity/channel/freq range used to generate the moment map ,as well as the channel width of the original cube.

### PR DESCRIPTION
**Pull request by markccchiang**
Closes #1024. Adding some information to the FITS header history about moment map generation. As shown below:

截圖 2022-04-28 下午1 22 13

### CODE REVIEW

**Comment by kswang1029** [comment]
@markccchiang quick questions, does this new info exist in the header of the in-memory moment images (the image auto appended to carta after the moment image is generated)? I am not sure where I can locate this new info. 🤔

I suggest we refine the extra moment generation info as the following:
1. add a string like CARTA MOMENT MAP GENERATOR LOG before add extra info from CARTA
2. the extra info should include
a. the input image for moment map generation
b. region info (pixel coordinate should be fine)
c. spectral range (channel, freq, and velocity). This is included already. 👍 
d. mask info

**Comment by markccchiang** [comment]
@kswang1029 as for the moment region info about pixel coordinates. Does it mean the control points of regions (e.g., point, rectangle, polygon..)?

**Comment by kswang1029** [comment]
> @kswang1029 as for the moment region info about pixel coordinates. Does it mean the control points of regions (e.g., point, rectangle, polygon..)?

yes.

**Comment by markccchiang** [comment]
Now the new moment map generation info from the header of moment images would look like this:

截圖 2022-05-03 下午1 55 45

**Comment by kswang1029** [comment]
> Now the new moment map generation info from the header of moment images would look like this:
> 
> 截圖 2022-05-03 下午1 55 45

this works nicely! Could you add region type too? Also we need rotation angle be part of the info.  

> HISTORY =  / Region control points: [402.9424, 374.0267], [242.3408, 294.6644] 

I suggest the format like this (please refer to the header of the exported tsv file from the spectral profiler)

rotbox[[402.942341pix, 374.026678pix], [242.340792pix, 294.664372pix], 0.000000deg]

**Review by kswang1029 — APPROVED** [review]
👍

**Comment by kswang1029** [comment]
@markccchiang just realized that the polarization info is missing. Could you add it too?

**Review by kswang1029 — CHANGES_REQUESTED** [review]
thanks for adding polarization info. It works with usual Stokes. Now CARTA supports computed polarization quantities too such as polarized intensity, angle, etc, so when these are selected, the displayed text is unknown.

HISTORY =  / Stokes: Unknown 

I suggest we call show this as 

Polarization: Stokes Q
Polarization: Fractional polarized intensity
Polarization: XY

**Comment by markccchiang** [comment]
> I suggest we call show this as
> 
> Polarization: Stokes Q 
> Polarization: Fractional polarized intensity 
> Polarization: XY

Please confirm are the full name of the computed stokes correct?

Ptotal: Total linearly polarized intensity
Plinear: Linearly polarized intensity
PFtotal: Total fractional linear polarization
PFlinear: Fractional linear polarization
Pangle: Linearly polarized angle

**Comment by kswang1029** [comment]
@markccchiang please consider

Ptotal: total polarization intensity
Plinear: linear polarization intensity
PFtotal: fractional total polarization intensity
PFlinear: fractional linear polarization intensity
Pangle: polarization angle

**Review by kswang1029 — APPROVED** [review]
looking good! 👍

File: src/Frame/Frame.cc [hunk 1]
```diff
@@ -2214,6 +2221,40 @@ bool Frame::GetStokesTypeIndex(const string& coordinate, int& stokes_index) {
...
+                break;
+            case CARTA::PolarizationType::PFlinear:
+                computed_stokes = "Fractional linear polarization intensity";
+                break;
+            case CARTA::PolarizationType::Pangle:
+                computed_stokes = "Polarization angle";
+                break;
+            default:
+                break;
+        }
```
Comment (confluence) [inline src/Frame/Frame.cc]: Could this be changed to a map lookup?

**Review by confluence — APPROVED** [review]

**Pull request merged**